### PR TITLE
DEP: Deprecate private namespaces in ``scipy.spatial``

### PR DESCRIPTION
**Pull request by czgdp1807**
#### Reference issue

gh-14360

#### What does this implement/fix?

I have moved the files as of now to private namespaces. Let me know if the changes are right (just leaving a thumbs up on this comment will also work for me). I will proceed with the code corrections then.

#### Additional information

cc: @rgommers

### CODE REVIEW

**Comment by czgdp1807** [comment]
Ignore the commit history. I am cleaning it up.

**Comment by czgdp1807** [comment]
> Ignore the commit history. I am cleaning it up.

Done. Let me know if the changes are good. I will add deprecation warnings tomorrow. @rgommers

**Comment by rgommers** [comment]
Looks fine to me. It would be better to do it all in one go @czgdp1807. I think you've got the hang of this, so no need for me to review intermediate changes I think:)

**Review by rgommers — APPROVED** [review]
LGTM now, thanks @czgdp1807!

File: scipy/spatial/transform/rotation.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    return __all__
+
+
+def __getattr__(name):
+    if name not in __all__:
+        raise AttributeError(
+            "scipy.spatial.transform.rotation is deprecated and has no attribute "
+            f"{name}. Try looking in scipy.spatial instead.")
+
+    warnings.warn(f"Please use `{name}` from the `scipy.spatial` namespace, "
```
Comment (swallan) [inline scipy/spatial/transform/rotation.py:28]: #15572 Should this warning indicate that these functions are part of `scipy.spatial.transform`?

Comment (rgommers) [reply scipy/spatial/transform/rotation.py:28]: Yes indeed, good catch @swallan

**Pull request merged**